src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+            // Enable Cycle Mode (AccelLow Power Mode):
+            // In PWR_MGMT_1 (0x6B) make CYCLE =1
+            WriteRegister(Register.PWR_MGMT_1, 0b0010_0000);
+            // Motion Interrupt Configuration Completed
+        }
+
+        /// <summary>
+        /// Initialize the MPU9250
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device</param>
+        /// <param name="autoDispose">Will automatically dispose the I2C device if true</param>
+        public Mpu9250(I2cDevice i2cDevice, bool autoDispose = true)
+        {
+            _i2cDevice = i2cDevice;
+            _autoDispose = autoDispose;
+            Reset();
+            PowerOn();
+            if (!CheckVersion())
+                throw new IOException($"This device does not contain the correct signature 0x71 for a MPU9250");
+            GyroscopeBandwidth = GyroscopeBandwidth.Bandwidth0250Hz;
+            GyroscopeRange = GyroscopeRange.Range0250Dps;
+            AccelerometerBandwidth = AccelerometerBandwidth.Bandwidth1130Hz;
+            AccelerometerRange = AccelerometerRange.Range02G;
+            // Setup I2C for the AK8963
+            WriteRegister(Register.USER_CTRL, (byte)UserControls.I2C_MST_EN);
+            // Speed of 400 kHz
+            WriteRegister(Register.I2C_MST_CTRL, (byte)I2cBussFrequency.Freqency400kHz);
+            _ak8963 = new Ak8963.Ak8963(_i2cDevice, new Ak8963Attached(), false);
+            Thread.Sleep(50);
+            if (!_ak8963.CheckVersion())
+            {
+                // Try to reset the device first
+                _ak8963.Reset();
+                // Wait a bit
+                if (!_ak8963.CheckVersion())
+                {
+                    // Try to reset the I2C Bus
+                    WriteRegister(Register.USER_CTRL, (byte)UserControls.I2C_MST_RST);
+                    Thread.Sleep(100);
+                    // Resetup again
+                    WriteRegister(Register.USER_CTRL, (byte)UserControls.I2C_MST_EN);
+                    WriteRegister(Register.I2C_MST_CTRL, (byte)I2cBussFrequency.Freqency400kHz);
+                    // Poorly documented time to wait after the I2C bus reset
+                    // Found out that waiting a little bit is needed. Exact time may be lower
+                    Thread.Sleep(100);
+                    // Try one more time
+                    if (!_ak8963.CheckVersion())
+                        throw new IOException($"This device does not contain the correct signature 0x48 for a AK8963 embedded into the MPU9250");
+                }
+            }
+        }
+
+        private void Reset()
+        {
+            WriteRegister(Register.PWR_MGMT_1, 0x80);
+            Thread.Sleep(100);
+            _wakeOnMotion = false;
+        }
+
+        private void PowerOn()
+        {
+            // this should be a soft reset
+            WriteRegister(Register.PWR_MGMT_1, 0x01);
+            Thread.Sleep(200);
```

Comment:
It makes me uneasy to  see all the places where we have Thread.Sleep in the code. Are all these waits documented anywhere?

Reply:
Adding in the code the reference and adjusting one of the timing from 200 to 100.